api/provider/aws/environment_create.go

```diff
@@ -81,39 +49,69 @@ func (e *EnvironmentProvider) Create(req models.CreateEnvironmentRequest) (strin
 		return "", err
 	}
 
-	launchConfigName := fqEnvironmentID
-	if err := e.createLC(
-		launchConfigName,
-		aws.StringValue(securityGroup.GroupId),
-		instanceType,
-		e.Config.InstanceProfile(),
-		e.Config.SSHKeyPair(),
-		amiID,
-		userData); err != nil {
-		return "", err
-	}
+	// creating asg, lc isn't required for dynamic environments
+	if strings.ToLower(req.EnvironmentType) != models.EnvironmentTypeDynamic {
+		var userDataTemplate []byte
+		var amiID string
+
+		switch strings.ToLower(req.OperatingSystem) {
+		case models.LinuxOS:
+			userDataTemplate = []byte(DefaultLinuxUserdataTemplate)
+			amiID = e.Config.LinuxAMI()
+		case models.WindowsOS:
+			userDataTemplate = []byte(DefaultWindowsUserdataTemplate)
+			amiID = e.Config.WindowsAMI()
+		default:
+			return "", fmt.Errorf("Operating system '%s' is not recognized", req.OperatingSystem)
+		}
 
-	maxScale := config.DefaultEnvironmentMaxScale
-	if req.MaxScale != 0 {
-		maxScale = req.MaxScale
-	}
+		if req.AMIID != "" {
+			amiID = req.AMIID
+		}
 
-	autoScalingGroupName := fqEnvironmentID
-	if err := e.createASG(
-		autoScalingGroupName,
-		launchConfigName,
-		int64(req.MinScale),
-		int64(maxScale),
-		e.Config.PrivateSubnets()); err != nil {
-		return "", err
+		instanceType := config.DefaultEnvironmentInstanceType
+		if req.InstanceType != "" {
+			instanceType = req.InstanceType
+		}
+
+		if len(req.UserDataTemplate) > 0 {
+			userDataTemplate = req.UserDataTemplate
+		}
+
+		userData, err := RenderUserData(fqEnvironmentID, e.Config.S3Bucket(), userDataTemplate)
```

Comment:
Will discuss in Slack, but for record keeping: how do we do private docker auth with dynamic environments?

Reply:
Yes.. I raised this issue previously as well. Private repo auth isn't supported in fargate as far as I'm aware.

The only solution I can see is modifying the task definition ourselves to inject a reference to an ECR image. We would also have to ensure aws permissions are setup to allow cross account access to an ECR.

Reply:
Glad this was brought up --- it might have been the only useful thing AWS shared w us on the Fargate call. They mentioned several things worth noting here: 

1) Fargate ONLY supports private repos via ECR
2) You have to use https://github.com/awslabs/amazon-ecr-credential-helper
3) ecr credential helper needs the following policies set, which I'm betting we need to add still in Layer0: https://docs.aws.amazon.com/AmazonECR/latest/userguide/ecr_managed_policies.html
3) Supplementary article on the ECR credential helper (sort of relevant) in this context: https://aws.amazon.com/blogs/compute/authenticating-amazon-ecr-repositories-for-docker-cli-with-credential-helper/